lib/eventsource.js

```diff
@@ -144,11 +230,9 @@ function EventSource (url, eventSourceInitDict) {
     }
 
     req = (isSecure ? https : http).request(options, function (res) {
-      self.connectionInProgress = false
       // Handle HTTP errors
-      if (res.statusCode === 500 || res.statusCode === 502 || res.statusCode === 503 || res.statusCode === 504) {
-        _emit('error', new Event('error', {status: res.statusCode, message: res.statusMessage}))
-        onConnectionClosed()
+      if (res.statusCode !== 200) {
```

Comment:
Are there other 200-range responses that could come back that will be interpreted as errors with this?

Reply:
Ah good question. I am not sure if the backend returns > 200 responses. In that case should I check for all 20x errors? Or fallback to the original condition.

Reply:
Have used the filter you provided me offline :D 

res.statusCode >= 400 && res.statusCode <= 599